### PR TITLE
terraform provider: bigquery dataset iam_member support

### PR DESCRIPTION
**Pull request by justinsb**
Manual cherry-pick of https://github.com/hashicorp/terraform-provider-google/pull/16322

### CODE REVIEW

File: pkg/test/resourcefixture/testdata/basic/bigquery/v1beta1/bigquerydatasetaccessblock/create.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Copyright 2024 Google LLC
```
Comment (maqiuyujoyce) [inline pkg/test/resourcefixture/testdata/basic/bigquery/v1beta1/bigquerydatasetaccessblock/create.yaml:1]: Could you re-structure the test cases based on our [guide](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/README.NewResourceFromTerraform.md#test-multiple-scenarios-for-the-same-resource-kind)? The expected folder structure will be:

```
pkg/test/resourcefixture/testdata/basic
├── bigquery
|   ├── v1beta1
|       ├── bigquerydataset
|         ├── basicbigquerydataset
|             └── create.yaml
|             └── update.yaml
|         ├── bigquerydatasetaccessblock
|             └── create.yaml
|             └── update.yaml
```

File: pkg/test/resourcefixture/testdata/basic/bigquery/v1beta1/bigquerydatasetaccessblock/_http.log [hunk 2]
```diff
@@ -0,0 +1,421 @@
+POST https://bigquery.googleapis.com/bigquery/v2/projects/${projectId}/datasets?alt=json
```
Comment (maqiuyujoyce) [inline pkg/test/resourcefixture/testdata/basic/bigquery/v1beta1/bigquerydatasetaccessblock/_http.log:1]: Remove?

**Review by maqiuyujoyce — COMMENTED** [review]
Thank you for cherrypicking the TF change and add a test case for it! Overall LGTM though some cleanup (?) may be needed before the merge.

File: pkg/test/resourcefixture/testdata/basic/bigquery/v1beta1/bigquerydatasetaccessblock/_http.log (hunk 2 above)
Comment (justinsb) [reply pkg/test/resourcefixture/testdata/basic/bigquery/v1beta1/bigquerydatasetaccessblock/_http.log:1]: Good call, will move to the mockgcp implementation for BigQuery

File: pkg/test/resourcefixture/testdata/basic/bigquery/v1beta1/bigquerydatasetaccessblock/create.yaml (hunk 1 above)
Comment (justinsb) [reply pkg/test/resourcefixture/testdata/basic/bigquery/v1beta1/bigquerydatasetaccessblock/create.yaml]: Done

**Review by tomasaschan — APPROVED** [review]
LGTM!

For our use case it would be interesting with a test that shows what happens if the binding is for `deleted:serviceAccount:...` but I don't see a reason it shouldn't work, given what is returned from upstream.

**Comment by justinsb** [comment]
> LGTM!
> 
> For our use case it would be interesting with a test that shows what happens if the binding is for `deleted:serviceAccount:...` but I don't see a reason it shouldn't work, given what is returned from upstream.


Thanks and agreed - it's a slightly more complicated test than we have done to date, but I am working on adding it to our testing arsenal in #1263

**Comment by justinsb** [comment]
@maqiuyujoyce I think it needs /approve, not just /lgtm

**Comment by maqiuyujoyce** [comment]
/approve

@justinsb why approve tag is also needed?

**Pull request merged**